workspace/__ardulib__/Graphics/Text.inl

```diff
@@ -0,0 +1,46 @@
+
+Text::Text(XGraphics* parent, char* text, size_t textLen)
+    : XGraphics(parent) {
+    _text = text;
+    _textLen = textLen;
+}
+
+void Text::setPosition(int16_t x, int16_t y) {
+    _pivot = XVector2<int16_t>(x, y);
+}
+
+void Text::setStyle(XColor* textColor) {
+    _textColor = textColor;
+}
+
+void Text::setStyle(uint8_t textSize) {
+    _textSize = textSize;
+}
+
+void Text::renderScanline(XRenderer* renderer, int16_t scanline, uint16_t* buffer, size_t bufferSize) {
+
+    if (scanline < _pivot.y || scanline > _pivot.y + 7 * _textSize - 1)
+        return;
+
+    uint16_t color = xcolorTo565(_textColor ? *_textColor : getFGColor());
+
+    for (uint8_t c = 0; c < _textLen; c++) {
+
+        for (uint8_t i = 0; i < 6; i++) {
+
+            uint8_t charLine = (i == 5) ? 0x00 : pgm_read_byte(font + (_text[c] * 5) + i);
+            for (uint8_t j = 0; j < 8; j++)
+                if (charLine & 1 << j) {
+
+                    int16_t cursorY = _pivot.y + j * _textSize;
+                    if (scanline >= cursorY && scanline < cursorY + _textSize) {
+
+                        int16_t cursorX = (c * 6 * _textSize) + _pivot.x + i * _textSize;
+                        for (int16_t x = cursorX; x < cursorX + _textSize; x++)
+                            if (x >= 0 && x < bufferSize)
+                                buffer[x] = color;
+                    }
+                }
+        }
+    }
```

Comment:
For each nesting level, please, add a comment on what’s going on.

To lower the nesting depth replace two big `if`’s with early `continue`:

```cpp
for (...) {
  if (foo) {
    ...
    if (bar) {
       ...
    }
  }
}

// to

for (...) {
  if (!foo) {
    // what happened?
    continue;
  }

  ...

  if (!bar) {
    // why do we continue?
    continue;
  }
  ...
}
```